CIP-????/README.md

```diff
@@ -0,0 +1,334 @@
+
+What we gain is the ability to specify and authorise this normal value flow in an out-of-order fashion. 
+Alice (whose transaction comes later in the value flow) can create their transaction first and send it to Bob (whose transaction comes earlier) and he can put them together into something valid.
+
+The linearizability requirement is the major driving difference between the design in this CIP and the one published in the "Babel Fees via Limited Liabilities" paper. 
+Both designs make use of the idea of “negative” inputs of one kind or another, but the liabilities’ design enables and embraces the possibility of loops in the value flow. 
+It is difficult to make this secure, which led to the painful restriction that minting policies be run in many cases where liabilities are used. 
+In practice that meant that many swap use cases could not be used with old tokens whose minting policies would not approve the new behaviour.
+
+Similarly, the liabilities design allowed easy access to arbitrary “flash loans” in Ada, which has often been a source of attacks on other blockchains.
+
+In contrast, if the value flow can be linearized then we know that nothing fundamentally new is possible here and we do not need to worry.
+
+#### Counterparty irrelevance
+
+Specifying transactions out-of-order allows us to leave more parts of a transaction unspecified. 
+We are used to not specifying where the outputs of transactions go, now we can not specify where the inputs of transactions come from.
+
+Not knowing where an output or input goes to gives you “counterparty irrelevance”: you don’t have to specify who is on the other side of that part of the trade. 
+Again, we are used to this with outputs. 
+We don’t in general know who is going to spend an output, although we generally restrict that with the credential on the output. 
+Unresolved inputs are just the same, but for inputs.
+
+Allowing counterparty irrelevance for inputs enables use cases where we expect someone to pay us, but we don’t know who. 
+An example of this is Gas Station Network-style fee sponsorship, where a dApp operator agrees to pay the script fees for its users. 
+Users could then create a transaction with an unresolved input covering the fees, which would later be resolved by the dApp operator.
+
+Having counterparty irrelevance for both inputs and outputs allows us to do trades where we both give and receive assets: e.g. a swap. 
+However, this presents us with a problem: we cannot resolve both the input and the output in one transaction, because this would violate linearizability. 
+The same transaction cannot appear both earlier and later in the value flow!
+
+We can solve this problem with atomic (indivisible) transaction groups: then we can use two transactions, one earlier and one later, but bind them together so that they cannot be split apart.
```

Comment:
Point out here that usually such indivisible transaction sets would come from a single user, who can sign them all without the need for additional communication